### ISSUE TITLE
Scaffold-DbContext filter with schemas and tables mixed not working

### ISSUE THREAD

**Issue by ProH4Ck** [oracle/dotnet-db-samples#73]
I have a huge Oracle 11 db with a lot of schemas/tables.
When I run Scaffold-DbContext passing -Schemas and -Tables parameters all tables in all schemas are generated; -Tables filter is ignored.

I made a test installation of Oracle 11 using official docker image and I created an empty DB with sample schemas and tables:

```SQL
CREATE USER SCHEMAA IDENTIFIED BY SCHEMAAPASSWORD;
/
GRANT CONNECT TO SCHEMAA;
/
GRANT CONNECT, RESOURCE, DBA TO SCHEMAA;
/
GRANT CREATE SESSION TO SCHEMAA;
/
CREATE USER SCHEMAB IDENTIFIED BY SCHEMABPASSWORD;
/
GRANT CONNECT TO SCHEMAB;
/
GRANT CONNECT, RESOURCE, DBA TO SCHEMAB;
/
GRANT CREATE SESSION TO SCHEMAB;
/
CREATE TABLE SCHEMAA.CUSTOMERS (
    ID NUMBER(10,0) NOT NULL,
    NAME VARCHAR2(250) NOT NULL,
    LASTNAME VARCHAR2(250) NOT NULL,
    PRIMARY KEY (ID)
)
/
CREATE TABLE SCHEMAB.EMPLOYEES (
    ID NUMBER(10,0) NOT NULL,
    NAME VARCHAR2(250) NOT NULL,
    LASTNAME VARCHAR2(250) NOT NULL,
    PRIMARY KEY (ID)
)
/
CREATE TABLE SCHEMAB.SALARIES (
    ID NUMBER(10,0) NOT NULL PRIMARY KEY,
    PAYDATE DATE NOT NULL,
    AMOUNT NUMBER(14,6) NOT NULL,
    CURRENCY VARCHAR2(50) NOT NULL,
    EMPLOYEE_ID NUMBER(10,0) NOT NULL,
    FOREIGN KEY (EMPLOYEE_ID) REFERENCES SCHEMAB.EMPLOYEES(ID)
)
/
```

Then, I run the following Scaffold-DbContext command:

```PS
Scaffold-DbContext "DATA SOURCE=(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=localhost)(PORT=1521))(CONNECT_DATA=(SID=XE)));PASSWORD=<<REDACTED>>;PERSIST SECURITY INFO=True;USER ID=SYSTEM" Oracle.EntityFrameworkCore -Force -Schemas SCHEMAA, SCHEMAB -Tables CUSTOMERS,EMPLOYEES
```

The expected result is that only CUSTOMERS and EMPLOYEES table are generated but when the command finishes execution also SALARIES appears in my .NET project

Since provider is not open sourced, to understand what's going wrong, I enabled sql tracing on db and probably I found the bugged query:

```SQL
SELECT t.table_name name, t.owner schema  FROM all_tables t WHERE t.table_name <> '__EFMigrationsHistory'  AND (t.owner IN (:s0, :s1)
OR t.table_name IN (:twiths0, :twiths1) AND CONCAT(t.owner, CONCAT(N'.', t.table_name)) IN (:sdott0, :sdott1))
```

Replacing parameters with values passed from command line:

```SQL
SELECT t.table_name name, t.owner schema  FROM all_tables t WHERE t.table_name <> '__EFMigrationsHistory'  AND (t.owner IN ('SCHEMAA', 'SCHEMAB')
OR t.table_name IN ('EMPLOYEES', 'CUSTOMERS') AND CONCAT(t.owner, CONCAT(N'.', t.table_name)) IN ('SCHEMAA.CUSTOMERS', 'SCHEMAB.EMPLOYEES'));
```

| NAME | SCHEMA |
| :--- | :--- |
| CUSTOMERS | SCHEMAA |
| EMPLOYEES | SCHEMAB |
| SALARIES | SCHEMAB |

I think to problem is ```OR``` condition that need to be switched to an ```AND```:

```SQL
SELECT t.table_name name, t.owner schema  FROM all_tables t WHERE t.table_name <> '__EFMigrationsHistory'  AND (t.owner IN ('SCHEMAA', 'SCHEMAB')
AND t.table_name IN ('EMPLOYEES', 'CUSTOMERS') AND CONCAT(t.owner, CONCAT(N'.', t.table_name)) IN ('SCHEMAA.CUSTOMERS', 'SCHEMAB.EMPLOYEES'));
```
| NAME | SCHEMA |
| :--- | :--- |
| EMPLOYEES | SCHEMAB |
| CUSTOMERS | SCHEMAA |

**Comment by ErikEJ** [comment]
@ProH4Ck Excellent debugging work!

**Comment by alexkeh** [comment]
@ProH4Ck I was able to reproduce the problem. I filed a bug (31469664) for my dev team to fix it.

Thanks for the very detailed test case, clear explanation, and debugging work!

**Comment by ProH4Ck** [comment]
> @ProH4Ck I was able to reproduce the problem. I filed a bug (31469664) for my dev team to fix it.
> 
> Thanks for the very detailed test case, clear explanation, and debugging work!

@alexkeh thanks for the fast reply. I searched in My Oracle Support with bug number but I didn't find it:

![image](https://user-images.githubusercontent.com/8057670/84249274-d6345d00-ab0a-11ea-9d44-d27a3ce82199.png)

**Comment by TomaszGrzmilas** [comment]
Hi,

When I use Scaffold-DbContext using Oracle.EntityFrameworkCore and pass a view name to '-Tables' parameter I get error "Unable to find a table in the database matching the selected table".

This is different behavior than using Microsoft.EntityFrameworkCore.SqlServer, where view can be pass as  '-Tables' parameter and it will be scaffolded.

If @ProH4Ck  found good query, all_tables don't have information about views. There should be checking in 'all_views' table too.

**Comment by alexkeh** [comment]
@TomaszGrzmilas Yes, this is a bug (31449920) in the beta that we're planning to fix by production.

**Comment by alexkeh** [comment]
@TomaszGrzmilas My team decided to release one more beta before production. It will fix most of the bugs that have been so far in the beta 1.

**Comment by TomaszGrzmilas** [comment]
@alexkeh 

Great to hear that :)

**Comment by alexkeh** [comment]
The Oracle EF Core 3.1 beta 2 will support the ability to define the specific tables to generate from multiple schemas. The syntax to use looks like the following:

`Scaffold-DbContext -Tables SCHEMA1.TABLEA, SCHEMA2.TABLEB, SCHEMA3.TABLEC`

**Comment by ErikEJ** [comment]
When?

**Comment by alexkeh** [comment]
Should be this week.

**Comment by alexkeh** [comment]
[Beta 2](https://www.nuget.org/packages/Oracle.EntityFrameworkCore/3.19.0-beta2) is now available. 

More [info about this beta release](https://medium.com/@alex.keh/oracle-ef-core-3-1-beta-2-now-available-254d306ff875).
